src/core/Akka/Event/LogEvent.cs

```diff
@@ -67,7 +67,23 @@ protected LogEvent()
         /// <returns>A <see cref="System.String" /> that represents this LogEvent.</returns>
         public override string ToString()
         {
-            return string.Format("[{0}][{1}][Thread {2}][{3}] {4}", LogLevel().ToString().Replace("Level", "").ToUpperInvariant(), Timestamp, Thread.ManagedThreadId.ToString().PadLeft(4, '0'), LogSource, Message);
+            return ToString("[{LogLevel}][{Timestamp}][Thread {ThreadId}][{LogSource}] {Message}");
+        }
+
+        public string ToString(string format)
+        {
+            format = format
+                .Replace("{LogLevel", "{0")
+                .Replace("{Timestamp", "{1")
+                .Replace("{ThreadId", "{2")
+                .Replace("{LogSource", "{3")
+                .Replace("{LogClass", "{4")
+                .Replace("{Message", "{5")
+                .Replace("{CrLf", "{6");
```

Comment:
Ok, CrLf might be silly, it could just as well be escaped in the template string